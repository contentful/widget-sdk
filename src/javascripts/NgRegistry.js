import angular from 'angular';

// See AngularInit.js
//
// angular.module('contentful/init').register is an object that contains
// functions to define the Angular modules below at runtime. These functions
// below allow registration within es6 files.
const register = (type) => (name, def) =>
  angular.module('contentful/init').register[type](name, def);

export const registerController = register('controller');
export const registerDirective = register('directive');
export const registerFilter = register('filter');
export const registerFactory = register('factory');
export const registerService = register('service');
export const registerConstant = register('constant');
export const registerProvider = register('provider');
export const registerValue = register('value');

export const initReady = () => {
  try {
    return angular.module('contentful/init').loaded === true;
  } catch (e) {
    return false;
  }
};

export const awaitInitReady = async () => {
  const ready = initReady();

  if (!ready) {
    await new Promise((resolve) => setTimeout(resolve, 100));

    return awaitInitReady();
  }

  return true;
};

export const appReady = () => {
  try {
    return angular.module('contentful/app').loaded === true;
  } catch (_) {
    return false;
  }
};

export const getModule = (name) => {
  try {
    return angular.module('contentful/init').getModule(name);
  } catch (e) {
    console.error(e); // eslint-disable-line no-console

    throw e;
  }
};
