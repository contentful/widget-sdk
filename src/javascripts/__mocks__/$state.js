export default {
  go: jest.fn(),
  href: jest.fn().mockImplementation(sref => `http://url-for-state-${sref}`)
};
