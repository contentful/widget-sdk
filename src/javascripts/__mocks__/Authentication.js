export const cancelUser = jest.fn();
