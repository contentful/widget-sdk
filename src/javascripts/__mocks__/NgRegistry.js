export const getModule = module => require(`ng/${module}`);
