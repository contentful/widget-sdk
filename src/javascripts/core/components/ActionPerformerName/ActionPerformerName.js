import React from 'react';
import { ActionPerformer } from '../ActionPerformer';

export const ActionPerformerName = ({ ...props }) => (
  <ActionPerformer {...props}>
    {({ formattedName, formattedNameAsString }) => (
      <span data-test-id="action-performer-name" title={formattedNameAsString}>
        {formattedName}
      </span>
    )}
  </ActionPerformer>
);
