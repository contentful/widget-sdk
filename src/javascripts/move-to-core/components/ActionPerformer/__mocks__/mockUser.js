export const mockUser = {
  firstName: 'Felix',
  lastName: 'Boenke',
  avatarUrl:
    'https://lh5.googleusercontent.com/-HeLJpAl8MH0/AAAAAAAAAAI/AAAAAAAAAAA/AKF05nBxeHRUNu-GDOkEb2POUqu49b9ChQ/s50/photo.jpg',
  email: 'felix.boenke@contentful.com',
  activated: true,
  signInCount: 19,
  confirmed: true,
  '2faEnabled': false,
  sys: {
    type: 'User',
    id: '12412312tasdasdaq1',
    version: 20,
    createdAt: '2020-03-02T14:14:03Z',
    updatedAt: '2020-04-17T09:20:49Z',
  },
};
