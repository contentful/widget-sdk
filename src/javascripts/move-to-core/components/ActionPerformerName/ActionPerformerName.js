import React from 'react';
import { ActionPerformer } from '../ActionPerformer';

// eslint-disable-next-line no-unused-vars
export const ActionPerformerName = ({ children, ...props }) => (
  <ActionPerformer {...props}>
    {({ formattedName }) => <span data-test-id="action-performer-name">{formattedName}</span>}
  </ActionPerformer>
);
