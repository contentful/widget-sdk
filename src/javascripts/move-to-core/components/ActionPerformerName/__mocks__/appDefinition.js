export const appDefinitionsMock = [
  {
    src: 'https://typeform.ctfapps.net/frontend',
    id: '1SZgLiOT63ijEzPJZQ4QWT',
    appDefinitionId: '1SZgLiOT63ijEzPJZQ4QWT',
    namespace: 'app',
    name: 'Typeform',
    fieldTypes: ['Symbol'],
    locations: ['app-config', 'entry-field', 'dialog'],
    appId: 'typeform',
    appIconUrl:
      '//images.ctfassets.net/lpjm8d10rkpy/6TFC6qGkkxgMe0QlQ72Lid/45bd015f693a58c71a5642c2280596a4/typeform-logo-symbol-1585307156_1.svg',
    sidebar: false,
    parameters: [],
    installationParameters: {
      definitions: [],
      values: {
        selectedWorkspaceId: 'asdasdasd',
      },
    },
  },
];
