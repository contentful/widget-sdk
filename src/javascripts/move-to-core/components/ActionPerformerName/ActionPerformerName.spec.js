import React from 'react';
import { getModule } from 'core/NgRegistry';
import { render } from '@testing-library/react';
import { ActionPerformerName } from './ActionPerformerName';
import { mockUser } from './__mocks__/mockUser';

jest.mock('core/NgRegistry', () => ({ getModule: jest.fn() }));

getModule.mockReturnValue({
  users: {
    get: () => Promise.resolve(mockUser),
  },
  user: mockUser,
});

describe('ActionPerformerName', () => {
  const link = {
    sys: {
      type: 'Link',
      linkType: 'User',
      id: mockUser.sys.id,
    },
  };

  it('should render the loading component initially', async () => {
    const { container } = render(<ActionPerformerName link={link} />);
    expect(container).toMatchSnapshot();
  });

  it('should render the basic variant with required props after loading', async () => {
    const { findByText } = render(<ActionPerformerName link={link} />);
    expect(await findByText('Me')).toMatchSnapshot();
  });

  it('should render a formatted variant of the name', async () => {
    const { findByText } = render(
      <ActionPerformerName
        link={link}
        formatName={(name) => `Published by ${name.toLowerCase()}`}
      />
    );
    expect(await findByText('Published by me')).toMatchSnapshot();
  });
});
