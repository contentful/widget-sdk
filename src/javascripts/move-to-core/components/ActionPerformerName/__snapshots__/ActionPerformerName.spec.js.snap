// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionPerformerName should render a formatted variant of the name 1`] = `
<span
  data-test-id="action-performer-name"
>
  Published by me
</span>
`;

exports[`ActionPerformerName should render the basic variant with required props after loading 1`] = `
<span
  data-test-id="action-performer-name"
>
  Me
</span>
`;

exports[`ActionPerformerName should render the loading component initially 1`] = `
<div>
  <div
    data-test-id="loading-variant"
  >
    Loading...
  </div>
</div>
`;
